src/App.tsx

```diff
@@ -19,55 +67,66 @@
               placeholder="Enter a part of the name"
               className="input"
               data-cy="search-input"
+              value={query}
+              onChange={handleQueryChange}
+              onFocus={() => {
+                setShowSuggestions(true);
+              }}
             />
           </div>
 
-          <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
-            <div className="dropdown-content">
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter de Decker</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Petronella de Decker</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Hercke</p>
+          {showSuggestions && filteredPeople.length > 0 && (
+            <div
+              className="dropdown-menu"
+              role="menu"
+              data-cy="suggestions-list"
+            >
+              <div className="dropdown-content">
+                {filteredPeople.map((people, index) => (
+                  <div
+                    className="dropdown-item"
+                    data-cy="suggestion-item"
+                    key={people.slug}
+                    onClick={() => {
+                      handleSuggestionClick(people);
+                    }}
+                    onMouseEnter={() => setHighlightedIndex(index)}
+                  >
+                    <p
+                      className={
+                        highlightedIndex === index
+                          ? 'has-text-link'
+                          : 'has-text-danger'
+                      }
+                    >
+                      {people.name}
+                    </p>
+                  </div>
+                ))}
```

Comment:
Avoid using the index as a key for list items. Instead, use a unique identifier from the `people` object, such as `people.slug`, to ensure keys are stable and unique. This follows the checklist requirement: 'don't generate key on render'.